app/src/main/java/org/wikipedia/readinglist/database/ReadingListDbHelper.java

```diff
@@ -240,12 +256,44 @@ void addPagesToList(@NonNull SQLiteDatabase db, @NonNull ReadingList list, @NonN
         return addedTitles;
     }
 
+    public List<String> movePagesToListIfNotExist(long sourceReadingListId, @NonNull ReadingList list, @NonNull List<PageTitle> titles) {
+        SQLiteDatabase db = getWritableDatabase();
+        db.beginTransaction();
+        List<String> movedTitles = new ArrayList<>();
+        try {
+            for (PageTitle title : titles) {
+                if (getPageByTitle(db, list.id(), title) != null) {
+                    continue;
+                }
+                movePageToList(db, sourceReadingListId, list, title);
+                movedTitles.add(title.getDisplayText());
+            }
+            db.setTransactionSuccessful();
+        } finally {
+            db.endTransaction();
+        }
+        if (movedTitles.isEmpty()) {
```

Comment:
Should this be `!movedTitles.isEmpty()`?

Reply:
Oops, you're right.